riscv-asm.md

```diff
@@ -214,7 +214,150 @@ ATTRIBUTE_VALUE := <string>
 
 #### `rvc`/`norvc`
 
-Enable/disable the C-extension for the following code region.
+This option will be deprecated soon after `.option arch` has been widely
+implemented on main stream open source toolchains.
+
+Enable/disable the C-extension for the following code region. This option is
+equivalent to `.option arch, +c`/`.option arch, -c`, but widely support by
+older toolchain versions.
+
+Alternative style:
+
+```
+.option push
+.option +c   # Alternative of .option rvc
+.option pop
+
+.option push
+.option -c   # Alternative of .option norvc
+.option pop
+```
+
+NOTE: .option rvc might set the ELF flag EF_RISCV_RVC in some toolchains. That
+might cause the linker to compress instructions in code regions where that was
+not intended.
+
+NOTE: There is a difference between `.option rvc`/`.option norvc` and
+`.option arch +c`/`.option arch -c`. The latter won't set EF_RISCV_RVC in the
+ELF flags.
+
+#### `arch`
+
+Enable and/or disable specific ISA extensions for the following code regions, but
+without changing the arch attribute and `EF_RISCV_RVC` in the ELF flags, that
+means it will not raise the minimal execution environment requirement, so the user
+should take care to the execution of the code regions around
+`.option push`/`.option arch`/`.option pop`.
+
+Syntax for `.option arch`:
+```
+
+.option arch, <EXTENSIONS-OR-FULLARCH>
+
+EXTENSIONS-OR-FULLARCH := <EXTENSIONS>
+                        | <FULLARCHSTR>
+
+EXTENSIONS             := <EXTENSION> ',' <EXTENSIONS>
+                        | <EXTENSION>
+
+FULLARCHSTR            := <full-arch-string>
+
+EXTENSION              := <OP> <EXTENSION-NAME> [<VERSION>]
+
+OP                     := '+'
+                        | '-'
+
+VERSION                := [0-9]+ 'p' [0-9]+
+                        | [1-9][0-9]*
```

Comment:
I'm not sure this is correct. Can you please walk me through it?

Reply:
I guess the current rule might be confused by `[`, `]`, so I try to rewrite the rule to allow VERSION as empty string, also expand first rule into two for forbid `0p0`.

```
EXTENSION              := <OP> <EXTENSION-NAME> [<VERSION>]

VERSION                := [0-9]+ 'p' [0-9]+
                        | [1-9][0-9]*
```

to

```
EXTENSION              := <OP> <EXTENSION-NAME> <VERSION>

VERSION                := [1-9][0-9]* 'p' [0-9]+
                        | [0-9]+ 'p' [1-9][0-9]*
                        | [1-9][0-9]*
                        |
```

Reply:
@luismarques let me know it's still unclear to you or any case you think it's still ambiguous

Reply:
LGTM now.